src/main/scala/edu/ie3/osmogrid/guardian/run/SubGridHandling.scala

```diff
@@ -82,371 +103,236 @@ trait SubGridHandling {
-        nodeMapping.values.toSet.asJava,
-        rawGridElements.getLines,
-        t2W.toSet.asJava,
-        t3W.toSet.asJava,
-        rawGridElements.getSwitches,
-        rawGridElements.getMeasurementUnits
-      )
-
-      new SubGridContainer(
-        grid.getGridName,
-        grid.getSubnet,
-        rawGrid,
-        grid.getSystemParticipants,
-        grid.getGraphics
-      )
+      case _ =>
+        // if no mv nodes are present
+        Map.empty
     }
-  }
 
-  private def assignSubnetNumbers(
-      subnets: Seq[SubGridContainer]
-  ): Try[Seq[SubGridContainer]] = Try {
-    subnets.zipWithIndex.map { case (subGrid, subnetNumber) =>
-      assignSubnetNumber(subGrid, subnetNumber + 1) match {
-        case Success(value) => value
-        case Failure(exception) =>
-          throw GridException(
-            s"Unable to update subgrid container '$subGrid'.",
-            exception
+    val hv = hvGrids.map { grids =>
+      if (grids.size == 1 && grids(0).getGridName == "dummyHvGrid") {
+
+        val hvNode: NodeInput = grids(0).getRawGrid.getNodes.asScala.toList
+          .sortBy(
+            _.getVoltLvl.getNominalVoltage.getValue.doubleValue()
           )
+          .lastOption
+          .getOrElse(throw GridException("No hv node found."))
+
+        (
+          Map(hvNode.getUuid -> hvNode.copy().subnet(hvOffset).build()),
+          hvOffset + 2
+        )
+      } else {
+        assignSubNetNumbers(grids, hvOffset)
       }
     }
+
+    // maps with updated nodes
+    val updateMap = lv.map(_._1).getOrElse(Map.empty) ++ mvCombined ++ hv
+      .map(_._1)
+      .getOrElse(Map.empty)
+
+    val allNodes: Seq[NodeInput] = List(lvGrids, mvGrids, hvGrids).flatten
+      .flatMap(_.flatMap(_.getRawGrid.getNodes.asScala))
+
+    // final update map
+    allNodes.map { node =>
+      node -> updateMap.getOrElse(node.getUuid, node)
+    }.toMap
   }
 
-  /** Assign a certain subnet number to the [[SubGridContainer]]. This is done
-    * by iterating through all elements and assigning the subnet number to nodes
-    * and updating the element reference, respectively.
-    *
-    * @param subGrid
-    *   The sub grid container to be updated
-    * @param subnetNumber
-    *   The sub net number to be assigned
+  /** Method to assign a sub grid number with an offset for nodes that are on a
+    * higher voltage level. Nodes with a lower voltage level are not updated
+    * @param subGrids
+    *   with nodes
+    * @param offset
+    *   for nodes
     * @return
-    *   The updated [[SubGridContainer]]
+    *   a map of updated nodes and an offset for nodes with a higher voltage
+    *   level
     */
-  private def assignSubnetNumber(
-      subGrid: SubGridContainer,
-      subnetNumber: Int
-  ): Try[SubGridContainer] = Try {
-    val nodes = updateNodes(
-      subGrid.getRawGrid.getNodes.asScala.toSeq,
-      subnetNumber
-    )
+  private def assignSubNetNumbers(
```

Comment:
I opt to change this method name since there is already one method assignSubNetNumbers doing other things.